src/activity_handlers.cpp

```diff
@@ -536,23 +536,34 @@ void activity_handlers::hotwire_finish( player_activity *act, player *)
 
 void activity_handlers::longsalvage_finish( player_activity *act, player *p )
 {
-    bool has_salvage_tool = p->has_items_with_quality( "CUT", 1, 1 );
-    if( !has_salvage_tool ) {
-        add_msg(m_bad, _("You no longer have the necessary tools to keep salvaging!"));
-    }
-
+    item &salvage_tool = p->i_at( act->values[2] );
```

Comment:
Yo, it looks like you broke longsalvage so hard, we still can't figure it out. For reference, longsalvage is "Cut up all you can". I'm not convinced you ever tested it.